tests/mqtt/test_adapter.py

```
@@ -0,0 +1,752 @@
+
+    def test_error_logged_if_broker_does_not_exist(self):
+        adapter = MQTTClientAdapter(
+            "some_company",
+            "test_car",
+            timeout=0.5,
+            broker_host="localhost",
+            port=1884,
+            event_queue=EventQueue(),
+        )
+        with self.assertLogs(logger=_logger._logger, level="ERROR"):
+            with self.assertRaises(ConnectionRefusedError):
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)